### PR TITLE
libpsl: don't depend on python when we cross compile

### PR DESCRIPTION
**Pull request by Mic92**
this is quite a big dependency that we don't want.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mic92** [comment]
@ofborg build pkgsStatic.curl curl

**Comment by Mic92** [comment]
@ofborg build pkgsCross.aarch64-multiplatform.curl

**Review by alyssais — CHANGES_REQUESTED** [review]
I don't think it makes sense to artificially restrict functionality that would otherwise work when cross compiling.  There's not any reason to assume that people want this functionality when doing native builds, but not when cross compiling, except when cross compiling to platforms where it shouldn't work.

I think this should be an option (so people can choose when avoid the big dependency, even in native cases), and it should default to only including the Python stuff in cases where that's possible, i.e. not when cross compiling to a platform Python doesn't support cross compiling to.  We can use the list in their configure script.  It might be nice if it was exposed in `python3`'s `passthru`, so that other packages could do the same thing without having to repeat the list.

File: pkgs/development/libraries/libpsl/default.nix [hunk 1]
```diff
@@ -12,6 +12,7 @@
 , python3
 , buildPackages
 , publicsuffix-list
+, enablePython3 ? python3.availableOn stdenv.hostPlatform
```
Comment (Mic92) [inline pkgs/development/libraries/libpsl/default.nix:15]: Here we go.

**Review by alyssais — CHANGES_REQUESTED** [review]
This won't work, because Python's meta.platforms lists platforms Python can be built for, but it can't encode that it can't be _cross compiled_ to those platforms.  For example, Python supports being built natively on NetBSD, but not being cross compiled to it.  We need a separate list of which platforms Python supports being cross compiled to.

File: pkgs/development/interpreters/python/cpython/default.nix [hunk 2]
```diff
@@ -655,6 +655,8 @@ in with passthru; stdenv.mkDerivation (finalAttrs: {
       nativeBuildInputs = with pkgsBuildBuild.python3.pkgs; [ sphinxHook python-docs-theme ];
     };
 
+    availableOn = platform: !platform.isStatic && lib.meta.availableOn platform self;
```
Comment (alyssais) [inline pkgs/development/interpreters/python/cpython/default.nix:658]: Python can be built statically.  Even if it didn't, we could encode that using `badPlatforms = [ lib.systems.inspect.platformPatterns.isStatic ]`, with no need to invent a new helper.

**Comment by Mic92** [comment]
> This won't work, because Python's meta.platforms lists platforms Python can be built for, but it can't encode that it can't be _cross compiled_ to those platforms. For example, Python supports being built natively on NetBSD, but not being cross compiled to it. We need a separate list of which platforms Python supports being cross compiled to.

Ok. I don't want to spend more time on fixing this. I will disable cross-compilation in the nix repository until someone else fixes this.

**Pull request closed without merging**